### PR TITLE
CUMULUS-1169 Retry throttled requests to the AWS StepFunctions API

### PR DESCRIPTION
**Pull request by yjpa7145**
Calls to the AWS StepFunctions API were encountering ThrottlingExceptions.

* Created a `@cumulus/common/StepFunctions` module to hold the functions created in this PR.  Note: I chose that name because of this rule: http://airbnb.io/javascript/#naming--PascalCase-singleton
* Added functions to query the StepFunctions API that support retries.
* Updated any code that called `describeExecution`, `describeStateMachine`, `getExecutionHistory`, or `listExecutions` to use those retry-enabled functions.
* Deprecated `@cumulus/common/step-functions`, which was not being used.
* Added `@cumulus/common/aws.retryOnThrottlingException()`, which will take a function and wrap it so that ThrottlingExceptions are retried.

### CODE REVIEW

File: packages/common/tests/test-StepFunctions.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const test = require('ava');
+const aws = require('../aws');
+const StepFunctions = require('../StepFunctions');
+const { throttleOnce } = require('../test-utils');
+
+test.serial('getExecutionHistory() retries if a ThrottlingException occurs', async (t) => {
+  const sfnBefore = aws.sfn;
+  try {
```
Comment (ifestus) [inline packages/common/tests/test-StepFunctions.js]: would it make sense to throw these setup blocks into a [beforeEach](https://github.com/avajs/ava/blob/master/docs/recipes/test-setup.md)?

File: packages/common/tests/test-aws.js [hunk 2]
```diff
@@ -149,3 +149,36 @@ test('pullStepFunctionEvent returns message from S3', async (t) => {
     stub.restore();
   }
 });
+
+const throwThrottlingException = () => {
```
Comment (ifestus) [inline packages/common/tests/test-aws.js]: Should this use the functions you've defined in `common/test-utils.js`?

File: packages/common/tests/test-aws.js [hunk 3]
```diff
@@ -149,3 +149,36 @@ test('pullStepFunctionEvent returns message from S3', async (t) => {
...
+  const throttlingException = new Error('ThrottlingException');
```
Comment (ifestus) [inline packages/common/tests/test-aws.js]: same as above

File: packages/common/tests/test-aws.js [hunk 4]
```diff
@@ -149,3 +149,36 @@ test('pullStepFunctionEvent returns message from S3', async (t) => {
...
+  throttlingException.code = 'ThrottlingException';
+
+  throw throttlingException;
+};
+
+const throttleOnce = (fn) => {
```
Comment (ifestus) [inline packages/common/tests/test-aws.js]: same as above

File: packages/common/tests/test-StepFunctions.js (hunk 1 above)
Comment (yjpa7145) [reply packages/common/tests/test-StepFunctions.js]: They're all slightly different (stubbing different `sfn` functions), but I could probably extract them out into a helper function.  I'll do that.

File: packages/common/tests/test-aws.js (hunk 2 above)
Comment (yjpa7145) [reply packages/common/tests/test-aws.js]: Good catch, yes.  I'll fix that.

File: packages/common/tests/test-StepFunctions.js (hunk 1 above)
Comment (yjpa7145) [reply packages/common/tests/test-StepFunctions.js]: Created a helper function.

File: packages/common/tests/test-aws.js (hunk 2 above)
Comment (yjpa7145) [reply packages/common/tests/test-aws.js]: Fixed.

File: packages/common/tests/test-aws.js (hunk 3 above)
Comment (yjpa7145) [reply packages/common/tests/test-aws.js]: Fixed.

File: packages/common/tests/test-aws.js (hunk 4 above)
Comment (yjpa7145) [reply packages/common/tests/test-aws.js]: Fixed.

**Comment by yjpa7145** [comment]
> Good changes 👍 - I just have some questions/suggestions.
> 
> Not sure if this is on purpose, but I found some areas where `sfn` is still used instead of `StepFunction`:
> 
>     * https://github.com/nasa/cumulus/blob/master/packages/api/tests/serial/es/test-es-indexer.js#L645
> 
>     * https://github.com/nasa/cumulus/blob/master/tasks/pdr-status-check/tests/index.js#L48

`describeExecution` is being stubbed in both of those tests, so there should never be a case where they cause a `ThrottlingException` that we need to worry about.  They're alright the way they are.

**Pull request merged**